src/toolchain.rs

```diff
@@ -465,7 +465,7 @@ pub struct CustomToolchain<'a>(&'a Toolchain<'a>);
 
 impl<'a> CustomToolchain<'a> {
     pub fn new(toolchain: &'a Toolchain<'a>) -> Result<CustomToolchain<'a>> {
-        if toolchain.is_custom() {
+        if toolchain.is_custom() && toolchain.name() != "none" {
```

Comment:
@rbtcollins Do you think we need to move this special name into `is_custom_name` fn? Or we can just determine it here?

Reply:
I think it's *not* a custom name, right? the whole point is that we're not allowing it as a name for custom toolchains.

Reply:
> I think it's _not_ a custom name, right?

Actually, in the current `is_custom_name` fn we treat it as a custom name. https://github.com/rust-lang/rustup/blob/060b8f9e3f5a8a810b49108aa57cee2e784ee104/src/toolchain.rs#L194

So it's more like a special case and probably we can put it into `is_custom_name`.

Reply:
I agree with @hi-rustin that this is perhaps the wrong place.

A quick look at the code shows that we haven't integrated the "none" value of a toolchain systematically into the code. We shouldn't need to patch it here and there and everywhere.

We have things like `resolve_toolchain(&name)` that might be a decent place to flat out prohibit 'none' as a toolchain name, irrespective of custom-or-not